rspec-tools/rspec_tools/validation/metadata.py

```diff
@@ -16,12 +16,25 @@ def get_json_schema():
   return json.loads(DEFAULT_SCHEMA_PATH.read_bytes())
 
 def validate_metadata(rule_language: LanguageSpecificRule):
+  validate_schema(rule_language)
+  validate_status(rule_language)
+
+def validate_schema(rule_language: LanguageSpecificRule):
   schema = get_json_schema()
   try:
     validate(instance=rule_language.metadata, schema=schema)
   except ValidationError as e:
     path = f'in {e.path[-1]}' if e.path else ''
     raise RuleValidationError(f'Rule {rule_language.id} has invalid metadata {path}: {e.message}') from e
 
+def validate_status(rule_language: LanguageSpecificRule):
+  status = rule_language.metadata.get('status')
+  if has_replacement_rules(rule_language) and status in ["beta", "ready"]:
+    raise RuleValidationError(f'Rule {rule_language.id} has invalid metadata:'
+      f' status can\'t be "{status}" for a rule with the replacement rule(s)')
```

Comment:
```suggestion
      f' status can\'t be "{status}" for a rule with replacement rule(s).')
```

(either remove `the`, or add the replacement rules in the error message `with the remplacement rule(s): SXXX, SYYY`)